goalc/compiler/compilation/Asm.cpp

```diff
@@ -319,6 +327,113 @@ Val* Compiler::compile_asm_svf(const goos::Object& form, const goos::Object& res
   return get_none();
 }
 
+void Compiler::check_vector_float_regs(const goos::Object& form,
+                                       Env* env,
+                                       std::vector<std::pair<std::string, RegVal*>> args) {
+  for (std::pair<std::string, RegVal*> arg : args) {
+    if (!arg.second->settable() || arg.second->ireg().reg_class != RegClass::VECTOR_FLOAT) {
+      throw_compiler_error(
+          form,
+          fmt::format("Invalid {} register for a vector float operation form. Got a {{}}.",
```

Comment:
The `throw_compiler_error` thing has formatting build in, I think this could be:
```cpp
throw_compiler_error(form, "Invalid {} register for a vector float operation form. Got a {}.", arg.first, arg.second->print());
```